Observer/Log.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Wtc\Film\Observer;
+
+use Psr\Log\LoggerInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Wtc\Film\Controller\Repository\Film;
+
+class Log implements ObserverInterface
+{
+    private Film $films;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        Film $films,
+        LoggerInterface $logger
+    ) {
+        $this->logger = $logger;
+        $this->films = $films;
+    }
+
+    public function execute(Observer $observer)
+    {
+        $films = $this->films->getFilmsFromRepository();
```

Comment:
Maybe you could get use the `getById` from the repository, instead of looping through the whole collection? That would need a bit of a rewrite of this method.